src/compass/rdr2geo.py

```diff
@@ -0,0 +1,103 @@
+
+    t_start = time.time()
+    info_channel.log("starting rdr2geo")
+
+    # common rdr2geo inits
+    dem_raster = isce3.io.Raster(cfg.dem)
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+
+    # check if gpu ok to use
+    use_gpu = isce3.core.gpu_check.use_gpu(cfg.gpu_enabled, cfg.gpu_id)
+    if use_gpu:
+        # Set the current CUDA device.
+        device = isce3.cuda.core.Device(cfg.gpu_id)
+        isce3.cuda.core.set_device(device)
+
+    for burst in cfg.bursts:
+        # get isce3 objs from burst
+        rdr_grid = burst.as_isce3_radargrid()
+        isce3_orbit = burst.orbit
+
+        # init grid doppler
+        grid_doppler = isce3.core.LUT2d()
+
+        # init CPU or CUDA object accordingly
+        if use_gpu:
+            Rdr2Geo = isce3.cuda.geometry.Rdr2Geo
+        else:
+            Rdr2Geo = isce3.geometry.Rdr2Geo
+
+        # init rdr2geo obj
+        rdr2geo_obj = Rdr2Geo(
+            rdr_grid,
+            isce3_orbit,
+            ellipsoid,
+            grid_doppler)
+
+        # set rdr2geo params
+        for key, val in cfg.rdr2geo_params.__dict__.items():
+            setattr(rdr2geo_obj, key, val)
+
+        # init output directory in product_path
+        output_path = f'{cfg.product_path}/{burst.burst_id}'
+        os.makedirs(output_path, exist_ok=True)
+
+        # prepare output rasters
+        topo_output = {'x':(True, gdal.GDT_Float64),
+                       'y':(True, gdal.GDT_Float64),
+                       'z':(True, gdal.GDT_Float64),
+                       'layoverShadowMask':(cfg.rdr2geo_params.compute_mask,
+                                            gdal.GDT_Byte)}
+        raster_list = [
+            isce3.io.Raster(f'{output_path}/{fname}.rdr', rdr_grid.width,
+                            rdr_grid.length, 1, dtype, 'ENVI')
+            if enabled else None
+            for fname, (enabled, dtype) in topo_output.items()]
+        x_raster, y_raster, z_raster, layover_shadow_raster = raster_list
+
+        # run rdr2geo
+        rdr2geo_obj.topo(dem_raster, x_raster, y_raster, z_raster,
+                         layover_shadow_raster=layover_shadow_raster)
+
+        # remove undesired/None rasters from raster list
+        raster_list = [raster for raster in raster_list if raster is not None]
+
+        # save non-None rasters to vrt
+        output_vrt = isce3.io.Raster(f'{output_path}/topo.vrt', raster_list)
+
+        output_vrt.set_epsg(epsg)
+
+    dt = time.time() - t_start
+    info_channel.log(f"rdr2geo successfully ran in {dt:.3f} seconds")
+
+
+if __name__ == "__main__":
+    '''run rdr2geo from command line'''
+    # load command line args
+    rdr2geo_parser = YamlArgparse()
+
+    # get a runconfig dict from command line args
+    rdr2geo_runconfig = RunConfig.load_from_yaml(rdr2geo_parser.args.run_config_path, 'rdr2geo')
+
+    # run rdr2geo
+    run(rdr2geo_runconfig)
```

Comment:
I think by the time we arrive here ```rdr2geo_runconfig.bursts``` includes all bursts from different polarizations in case if the polarization is dual-pol. Is that right? 
If yes then we will be repeating the same computation of rdr2geo twice (for two pols of same burst) which is really unnecessary and expensive. We need to run rdr2geo only once. 
Similarly later we should run geo2rdr also one for two polarizations of the same burst. Resample will need to be run on each polarization though.

Reply:
Addressed with bc0ec83

Reply:
Thank you @LiangJYu I think the polarization configuration from schema to evaluations in different steps looks good now. 
However, I don't think if this specific comment here has been addressed. Let's say we have only one burst (one burst ID) in the run config but polarization is set to "dual-pol". In this case "rdr2geo_runconfig.bursts" will be a list of two bursts (same burst ID but two different polarizations). Do you agree?

If correct then by running run(rdr2geo_runconfig) we will be running rdr2geo twice, which is unnecessary because rdr2geo output for the same burst ID is the same for two different polarizations. The efficient way would be to run rdr2geo only one time for the two polarizations of the two bursts.

From an end to end workflow how shall we deal with it. One easy solution can be to check for polarization before calling  RunConfig.load_from_yaml(...) and force only one polarization to load_from_yaml when the function is running for rdr2geo and geo2rdr. 

Any other better solution?

Reply:
I will rework. Reverting bc0ec83

Reply:
After reviewing bc0ec83, I think it the implementation is mostly correct.

Bursts in a runconfig object are now [stored as](https://github.com/LiangJYu/COMPASS/blob/bc0ec831fa85d32940e5639235354464c8361fa9/src/compass/utils/runconfig.py#L127-L137) `list[list[Sentinel1BurstSlc]]` where each `list[Sentinel1BurstSlc]` have different polarizations and share a common burst ID . This implementation is missing rdr2geo specific logic to check that bursts of the duplicate ID+polarization do not get appended.

Change above impacts `rdr2geo.py` by:
1. [iterating](https://github.com/LiangJYu/COMPASS/blob/bc0ec831fa85d32940e5639235354464c8361fa9/src/compass/rdr2geo.py#L38) over a `list[list[Sentinel1BurstSlc]]` instead of `list[Sentinel1BurstSlc]`.
2. first [writing SLC to ENVI](https://github.com/LiangJYu/COMPASS/blob/bc0ec831fa85d32940e5639235354464c8361fa9/src/compass/rdr2geo.py#L44-L45) for up to two polarizations by iterating over a `list[Sentinel1BurstSlc]`
3. then [running rdr2geo](https://github.com/LiangJYu/COMPASS/blob/bc0ec831fa85d32940e5639235354464c8361fa9/src/compass/rdr2geo.py#L47-L93) once with the last `Sentinel1BurstSlc` from write to ENVI iteration above.